packages/query-core/src/queryClient.ts

```diff
@@ -68,13 +65,12 @@ export class QueryClient {
   constructor(config: QueryClientConfig = {}) {
     this.queryCache = config.queryCache || new QueryCache()
     this.mutationCache = config.mutationCache || new MutationCache()
-    this.logger = config.logger || defaultLogger
     this.defaultOptions = config.defaultOptions || {}
     this.queryDefaults = []
     this.mutationDefaults = []
 
-    if (process.env.NODE_ENV !== 'production' && config.logger) {
-      this.logger.error(
+    if (process.env.NODE_ENV !== 'production') {
+      console.error(
         `Passing a custom logger has been deprecated and will be removed in the next major version.`,
       )
     }
```

Comment:
this is a deprecation warning for v4 that isn't necessary with v5 anymore. Please remove it.